packages/svelte/src/lib/chat.svelte.ts

```diff
@@ -0,0 +1,341 @@
+import {
+  fillMessageParts,
+  generateId,
+  type UIMessage,
+  type UseChatOptions,
+  type JSONValue,
+  type ChatRequest,
+  extractMaxToolInvocationStep,
+  callChatApi,
+  shouldResubmitMessages,
+  type Message,
+  type CreateMessage,
+  type ChatRequestOptions,
+  prepareAttachmentsForRequest,
+  getMessageParts,
+  updateToolCallResult,
+  isAssistantMessageWithCompletedToolCalls,
+} from "@ai-sdk/ui-utils";
+import { isAbortError } from "@ai-sdk/provider-utils";
+import {
+  KeyedChatStore,
+  getChatContext,
+  hasChatContext,
+} from "./chat-context.svelte.js";
+
+export type ChatOptions = Readonly<
+  Omit<UseChatOptions, "keepLastMessageOnError"> & {
+    /**
+     * Maximum number of sequential LLM calls (steps), e.g. when you use tool calls.
+     * Must be at least 1.
+     * A maximum number is required to prevent infinite loops in the case of misconfigured tools.
+     * By default, it's set to 1, which means that only a single LLM call is made.
+     * @default 1
+     */
+    maxSteps?: number;
+  }
+>;
+
+export type { CreateMessage, Message, UIMessage };
+
+export class Chat {
+  readonly #options: ChatOptions = {};
+  readonly #api = $derived(this.#options.api ?? "/api/chat");
+  readonly #generateId = $derived(this.#options.generateId ?? generateId);
+  readonly #id = $derived(this.#options.id ?? this.#generateId());
+  readonly #maxSteps = $derived(this.#options.maxSteps ?? 1);
+  readonly #streamProtocol = $derived(this.#options.streamProtocol ?? "data");
+  readonly #keyedStore = $state<KeyedChatStore>()!;
+  readonly #store = $derived(this.#keyedStore.get(this.#id));
```

Comment:
This is the chat store at this ID. The `KeyedStore` class will automatically create a store for a given ID if it doesn't exist yet.